src/workerd/jsg/jsg-test.c++

```diff
@@ -423,31 +423,33 @@ KJ_TEST("jsg::Lock getUuid") {
 KJ_TEST("External memory adjustment") {
   IsolateUuidIsolate isolate(v8System, kj::heap<IsolateObserver>());
   isolate.runInLockScope([&](IsolateUuidIsolate::Lock& lock) {
-    // Creating with a specific amount works as expected
-    auto adjuster = lock.getExternalMemoryAdjustment(100);
-    KJ_ASSERT(adjuster.getAmount() == 100);
-
-    // Adjusting up works as expected
-    adjuster.adjust(lock, 10);
-    KJ_ASSERT(adjuster.getAmount() == 110);
-
-    // Adjusting down works as expected
-    adjuster.adjust(lock, -10);
-    KJ_ASSERT(adjuster.getAmount() == 100);
-
-    // Setting an explicit value just works
-    adjuster.set(lock, 50);
-    KJ_ASSERT(adjuster.getAmount() == 50);
-
-    // Decrementing by more than the amount just sets to 0
-    adjuster.adjust(lock, -200);
-    KJ_ASSERT(adjuster.getAmount() == 0);
-
-    adjuster.set(lock, 100);
-    auto adjuster2 = kj::mv(adjuster);
-    KJ_ASSERT(adjuster2.getAmount() == 100);
-    KJ_ASSERT(adjuster.getAmount() == 0);
-
+    // Creating an inner scope to check the case where the adjustment object does not outlive the isolate
+    {
+      // Creating with a specific amount works as expected
+      auto adjuster = lock.getExternalMemoryAdjustment(100);
+      KJ_ASSERT(adjuster.getAmount() == 100);
+
+      // Adjusting up works as expected
+      adjuster.adjust(10);
+      KJ_ASSERT(adjuster.getAmount() == 110);
+
+      // Adjusting down works as expected
+      adjuster.adjust(-10);
+      KJ_ASSERT(adjuster.getAmount() == 100);
+
+      // Setting an explicit value just works
+      adjuster.set(50);
+      KJ_ASSERT(adjuster.getAmount() == 50);
+
+      // Decrementing by more than the amount just sets to 0
```

Comment:
This seems really error prone
Why not throw an exception? We should find out when our code messes up memory accounting

Reply:
@jasnell Any objections if I switch to an exception?

Reply:
+1 on throwing

Reply:
No objection